lib/sqlvalue/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+    }
+}
+
+impl From<f32> for SqlValue {
+    fn from(value: f32) -> Self {
+        SqlValue::Flt(value)
+    }
+}
+
+impl From<f64> for SqlValue {
+    fn from(value: f64) -> Self {
+        SqlValue::Dbl(value)
+    }
+}
+
+impl From<OrderedFloat<f32>> for SqlValue {
+    fn from(value: OrderedFloat<f32>) -> Self {
+        SqlValue::Flt(value.into())
+    }
+}
+
+impl From<OrderedFloat<f64>> for SqlValue {
+    fn from(value: OrderedFloat<f64>) -> Self {
+        SqlValue::Dbl(value.into())
+    }
+}
+
+impl From<String> for SqlValue {
+    fn from(value: String) -> Self {
+        SqlValue::Str(value)
+    }
+}
+
+impl From<Option<i32>> for SqlValue {
+    fn from(value: Option<i32>) -> Self {
+        SqlValue::OptInt(value)
+    }
+}
+
+impl From<Option<bool>> for SqlValue {
+    fn from(value: Option<bool>) -> Self {
+        SqlValue::OptBool(value)
+    }
+}
+
+impl From<Option<f32>> for SqlValue {
+    fn from(value: Option<f32>) -> Self {
+        SqlValue::OptFlt(value)
+    }
+}
+
+impl From<Option<f64>> for SqlValue {
+    fn from(value: Option<f64>) -> Self {
+        SqlValue::OptDbl(value)
+    }
+}
+
+impl From<Option<OrderedFloat<f32>>> for SqlValue {
+    fn from(value: Option<OrderedFloat<f32>>) -> Self {
+        match value {
+            None => SqlValue::OptFlt(None),
+            Some(OrderedFloat(x)) => SqlValue::OptFlt(Some(x)),
+        }
+    }
+}
+
+impl From<Option<OrderedFloat<f64>>> for SqlValue {
+    fn from(value: Option<OrderedFloat<f64>>) -> Self {
+        match value {
+            None => SqlValue::OptDbl(None),
+            Some(OrderedFloat(x)) => SqlValue::OptDbl(Some(x)),
+        }
+    }
+}
+
+impl From<Option<String>> for SqlValue {
+    fn from(value: Option<String>) -> Self {
+        SqlValue::OptStr(value)
+    }
+}
+
+pub struct SqlRow {
+    values: Vec<SqlValue>,
+}
+
+pub trait ToSqlRow {
+    fn to_row(&self) -> SqlRow;
+}
+
+impl SqlRow {
+    pub fn new() -> Self {
+        SqlRow { values: Vec::default() }
+    }
+
+    // Output the SqlRow value in the format expected by the tests
+    // in SqlLogicTest.
+    // 'format' is a string with characters I, R, or T, standing
+    // respectively for integers, real, or text.
+    pub fn to_slt_strings(self, format: &String) -> Vec<String> {
+        if self.values.len() != format.len() {
+            panic!("Mismatched format {} vs len {}", format.len(), self.values.len())
+        }
+        let mut result = Vec::<String>::new();
+        for elem in self.values.iter().zip(format.chars()) {
+            result.push(elem.0.format_slt(&elem.1));
+        }
+        result
+    }
+}
+
+impl SqlRow {
+    pub fn push(self: &mut Self, value: SqlValue) {
+        self.values.push(value)
+    }
+}
+
+impl Default for SqlRow {
+    fn default() -> Self {
+        SqlRow { values: Vec::default() }
+    }
+}
+
+pub trait SqlLogicTestFormat {
+    fn format_slt(&self, arg: &char) -> String;
+}
+
+
+fn print_string(s: &String) -> String {
```

Comment:
not obvious to me what this does on first glance, maybe add some comment to function